push/service.go

```diff
@@ -2,20 +2,47 @@ package push
 
 import (
 	"encoding/json"
+	"fmt"
 
 	"github.com/RobotsAndPencils/buford/payload"
 	"github.com/RobotsAndPencils/buford/push"
 	"github.com/pkg/errors"
 	"golang.org/x/net/context"
+
+	"github.com/micromdm/micromdm/pubsub"
+	"github.com/micromdm/micromdm/queue"
 )
 
 type Push struct {
 	db      *DB
 	pushsvc *push.Service
 }
 
-func New(db *DB, push *push.Service) *Push {
-	return &Push{db, push}
+func New(db *DB, push *push.Service, sub pubsub.Subscriber) (*Push, error) {
+	pushSvc := Push{db, push}
+
+	commandQueuedEvents, err := sub.Subscribe("push-info", queue.CommandQueuedTopic)
+	if err != nil {
+		return nil, errors.Wrapf(err,
+			"subscribing push to %s topic", queue.CommandQueuedTopic)
+	}
+	go func() {
+		for {
+			select {
+			case event := <-commandQueuedEvents:
+				udid, _, err := queue.UnmarshalQueuedCommand(event.Message)
+				if err != nil {
+					fmt.Println(err)
+				}
+				_, err = pushSvc.Push(nil, udid)
```

Comment:
One of the things I'd like this code to do is to _not_ send push notifications when a device enrolls. If the device is just enrolling it will continuously check in to the server until we return no more payloads. 
We'll want to handle pushing notifications to the device only after enrollment. 
Maybe each device get's it's own queue/subscription and we have Start() and Stop() methods on that queue?

Of course what I'm proposing is hard, so feel free to punt on that for now.